pulp_2to3_migration/app/plugin/rpm/migrator.py

```diff
@@ -271,11 +271,15 @@ def relate_packages_to_group(self, group_dc):
         packages = group_dc.content.packages
         package_list = [pkg['name'] for pkg in packages]
         pulp2_repo_id = group_dc.extra_data.get('pulp2_repo_id')
-        pulp2_repo = Pulp2Repository.objects.get(pulp2_repo_id=pulp2_repo_id)
-        # all pulp2 unit_ids for rpm within the pulp2repo
-        unit_ids = Pulp2RepoContent.objects.filter(
-            pulp2_repository=pulp2_repo,
-            pulp2_content_type_id='rpm').values_list('pulp2_unit_id', flat=True).iterator()
+        pulp2_repo = Pulp2Repository.objects.filter(pulp2_repo_id=pulp2_repo_id).first()
+        # if pulp2_repo is found, find all pulp2 unit_ids for rpm/srpm within the pulp2_repo
+        # otherwise find all pulp2 unit_ids for rpm/srpm content
+        filters = {}
+        if pulp2_repo:
+            filters['pulp2_repository'] = pulp2_repo
```

Comment:
I'm not sure if it's right to search all packages if we don't have a repo.
Can we prevent migration of such content earlier, so here we definitely gave a repo migrated?

Reply:
this was the workaround in case content-only migration is performed. if we plan to skip content migration that depends on the repo and repo is not migrated (https://pulp.plan.io/issues/6511) i am cool to remove this.
Also we need to fix this issue because no matter what repos are specified in plan, all content is still migrated https://pulp.plan.io/issues/5976

Reply:
https://pulp.plan.io/issues/5976, we discussed it in the past, that it's more a feature than a bug and it's probably better to keep it this way, otherwise we won't be able to rely on the timestamps.
For people who want to migrate the majority or all of repos, it's ok.
For those who wants to migrate just few repos, it's probably easier to recreate those repos in pulp3 and not be bothered with the migration plugin.